src/libs/Middleware/HandleUnusedOptimisticID.ts

```diff
@@ -10,6 +10,10 @@ const handleUnusedOptimisticID: Middleware = (requestResponse, request, isFromSe
         const responseOnyxData = response?.onyxData ?? [];
         responseOnyxData.forEach((onyxData) => {
             const key = onyxData.key;
+            if (!key) {
+                return;
+            }
+
             if (!key.startsWith(ONYXKEYS.COLLECTION.REPORT)) {
```

Comment:
FYI optional chaining is allowed in TS:
```suggestion
            if (!key?.startsWith(ONYXKEYS.COLLECTION.REPORT)) {
```